plugin.php

```diff
@@ -58,18 +58,53 @@ function () {
 
 /* class files get loaded by the autoloader */
 
+global $newsletter_builder_email_provider;
+
 /**
  * Instantiate the plugin.
  */
 function main() {
 	new Ads();
 	new Breaking_Recipients();
-	Campaign_Monitor_Client::instance()->setup();
 	new Email_Types();
+	new Settings();
 	new Media();
 	new WP_Newsletter_Builder();
 	new Rest_API_Endpoints();
 	new Rest_API_Fields();
 	new Rest_API_Query();
+	// Find selected email provider and instantiate it.
+	$selected_email_provider = apply_filters( 'wp_newsletter_builder_selected_provider', '' );
+
+	// Check if provider has been selected and exists.
+	if ( empty( $selected_email_provider ) || ! class_exists( $selected_email_provider ) ) {
+		\add_action(
+			'admin_notices',
+			function () {
+				?>
+				<div class="notice notice-error">
+					<p>
+						<?php
+						printf(
+							// translators: %s is the filter name.
+							esc_html__(
+								'No email provider selected for WP Newsletter Builder. Use the %s filter to specify one.',
+								'wp-newsletter-builder'
+							),
+							'<code>wp_newsletter_builder_selected_provider</code>'
+						);
+						?>
+					</p>
+				</div>
+				<?php
+			}
+		);
+
+		return;
+	}
+	global $newsletter_builder_email_provider;
+	$provider                          = new $selected_email_provider();
```

Comment:
To my taste adding a blank line below this so the formatter doesn't add so many spaces is more readable.